src/main/java/mate/academy/spring/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.spring.service.impl;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserService userService;
+
+    public UserDetailsServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
```

Comment:
you can simplify  it 
```suggestion
      User user = userService.findByEmail(email).orElseThrow(() -> 
           throw new UsernameNotFoundException("User not found by email " + email)
      );
```